module/VuFind/src/VuFind/Search/Base/DateRangeOptionsInterface.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * Interface for date range options
+ *
+ * Copyright (C) The National Library of Finland 2025.
+ *
+ * PHP version 8
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search_Base
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Search\Base;
+
+/**
+ * Interface for date range options
+ *
+ * @category VuFind
+ * @package  Search_Base
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+interface DateRangeOptionsInterface
+{
+    /**
+     * Get date range facets.
+     *
+     * @return array
+     */
+    public function getDateRangeFacets(): array;
+
+    /**
+     * Get full date range facets.
```

Comment:
Should we expand this comment to clarify exactly what "full date range" actually means? Even though I probably created the term in the first place, it was not immediately clear to me what it meant.

```suggestion
     * Get full (day/month/year, rather than just year) date range facets.
```